enhancements/ingress/empty-requests-policy.md

```diff
@@ -0,0 +1,267 @@
+authors:
+  - "@Miciah"
+reviewers:
+  - "@danehans"
+  - "@frobware"
+  - "@knobunc"
+  - "@sgreene570"
+approvers:
+  - "@knobunc"
+creation-date: 2020-08-31
+last-updated: 2020-08-31
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Ingress Empty Requests Policy
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement extends the IngressController API to allow the user to
+configure IngressControllers to log or ignore "empty requests", meaning
+incoming connections on which the IngressController receives no requests.
+
+## Motivation
+
+Empty requests typically result from load balancers' health probes or Web
+browsers' speculative connections ("preconnect").  Administrators might consider
+such connections to be noise that clutters logs and wastes resources.
+
+However, empty requests may also be caused by network errors, in which case
+logging the requests may be useful for diagnosing the errors.  In addition, the
+connections may be caused by port scans, in which case logging the connections
+may aid in detecting intrusion attempts.
+
+Whether or not these connections should be logged or ignored depends on the
+environment and the preferences of the cluster administrator, and thus an API to
+specify the desired configuration is needed.
+
+### Goals
+
+1. Enable the cluster administrator to specify whether an IngressController should log empty requests.
+2. Enable the cluster administrator to specify whether an IngressController should respond with HTTP 408 when an empty request times out.
+3. Enable the cluster administrator to specify whether an IngressController should count empty requests in metrics.
+
+### Non-Goal
+
+## Proposal
+
+To enable the cluster administrator to configure logging of empty requests, the
+IngressController API is extended by adding an optional `LogEmptyRequests` field
+with type `LoggingPolicy` to `AccessLogging`:
+
+```go
+type AccessLogging struct {
+	// ...
+
+	// logEmptyRequests specifies how connections on which no request is
+	// received should be logged.  Typically, these empty requests come from
+	// load balancers' health probes or Web browsers' speculative
+	// connections ("preconnect"), in which case logging these requests may
+	// be undesirable.  However, these requests may also be caused by
+	// network errors, in which case logging empty requests may be useful
+	// for diagnosing the errors.  In addition, these requests may be caused
+	// by port scans, in which case logging empty requests may aid in
+	// detecting intrusion attempts.  Allowed values for this field are
+	// "Log" and "Ignore".  If the field is empty. empty requests are
+	// logged.
```

Comment:
consider using defaults. Makes it much easier to discover.

Reply:
Done.  Thanks!